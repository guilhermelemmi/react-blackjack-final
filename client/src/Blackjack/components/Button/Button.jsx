import "./Button.css";

function Button({ children, type, onClick, ...otherProps }) {
  return (
    <button className={`button ${type}`} onClick={onClick} {...otherProps} >
      {children}
    </button>
  );
}

export default Button;
