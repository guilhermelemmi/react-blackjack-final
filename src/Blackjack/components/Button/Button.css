button {
  display: inline-block;
  border: none;
  padding: 1rem 2rem;
  margin: 0;
  text-decoration: none;
  color: #fefefe;
  cursor: pointer;
  text-align: center;
  font-size: 1.2rem;
  border: 1px solid #eee;
}

button:hover,
button:focus {
  border: 1px solid #fff;
}

button:focus {
  outline: 1px solid #fff;
  outline-offset: -4px;
}

.primary {
  background-color: #0b4d00;
}

.primary:hover {
  background-color: #049b22;
}

.secondary {
  margin-right: 4px;
  background-color: #854703;
}

.secondary:hover {
  background-color: #e78f28;
}
