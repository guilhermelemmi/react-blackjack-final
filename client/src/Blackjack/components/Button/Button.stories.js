import Button from "./Button";

export default {
  component: Button,
};

export const Primary = {
  render: () => <Button type="primary">Click me</Button>,
};

export const Secondary = {
  render: () => <Button type="secondary">Click me</Button>,
};
