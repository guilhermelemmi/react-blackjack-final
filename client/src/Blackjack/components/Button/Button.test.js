import { render, screen, fireEvent } from "@testing-library/react";
import Button from "./Button";

const noop = () => {};

describe("Button", () => {
  test("Button onClick works as expected", async () => {
    const handleClick = jest.fn();
    render(<Button onClick={handleClick}>Click Me</Button>);
    fireEvent.click(screen.getByText(/Click me/i));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  test("Primary button matches snapshot", async () => {
    const view = render(
      <Button type="primary" onClick={noop}>
        Click me
      </Button>,
    );
    expect(view.container).toMatchSnapshot();
  });

  test("Secondary button matches snapshot", async () => {
    const view = render(
      <Button type="secondary" onClick={noop}>
        Click me
      </Button>,
    );
    expect(view.container).toMatchSnapshot();
  });
});
