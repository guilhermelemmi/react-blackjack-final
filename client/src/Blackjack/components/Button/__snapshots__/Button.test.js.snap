// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Primary button matches snapshot 1`] = `
<div>
  <button
    class="button primary"
  >
    Click me
  </button>
</div>
`;

exports[`Button Secondary button matches snapshot 1`] = `
<div>
  <button
    class="button secondary"
  >
    Click me
  </button>
</div>
`;
